File: joao/1608_special_array_with_x_elements_greater_than_or_equal_x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        
        left, right = 0, n
        
        while left <= right:
            mid = (left + right) // 2
            count = sum(1 for num in nums if num >= mid)
            
            if count == mid:
                return mid
            elif count < mid:
                right = mid - 1
            else:
                left = mid + 1
        
        return -1
        

File: marcellevargas/1608_Special_Array_With_X_Elements_Greater_Than_or_Equal_X.js
function specialArray(nums) {
  nums.sort((a, b) => a - b);
  for (let x = 0; x <= nums.length; x++) {
    let count = 0;
    for (let i = 0; i < nums.length; i++) {
      if (nums[i] >= x) {
        count++;
      }
    }
    if (count === x) {
      return x;
    }
  }
  return -1;
}

console.log(specialArray([3, 9, 7, 8, 3, 8, 6, 6]));